analytics.js

```diff
@@ -167,6 +170,25 @@ class Analytics {
       if (typeof response === "string") {
         response = JSON.parse(response);
       }
+
+      // Fetch Error reporting enable option from sourceConfig
+      const IsErrorReportEnabled = get(
+        response.source.config,
+        "statsCollection.errorReports.enabled"
+      );
+
+      // Load Bugsnag only if it is enabled in the source config
+      if (IsErrorReportEnabled === true) {
+        // Fetch the name of the Error reporter from sourceConfig
+        const provider = get(
+          response.source.config,
+          "statsCollection.errorReports.provider"
+        );
+        if (provider === "bugsnag") {
```

Comment:
Also, validate the provider value for allowed providers.

Reply:
Addressed